### PR TITLE
Fix logging of user name in historical logs

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12796

When a historical log entry is made, if the user has their preferences set to show GLPI IDs, the logged name has the ID shown twice. I don't think the name order or ID display should be dependent on the individual user's preference since the data is statically logged (not computed when displayed). I don't think it is good to rely on the defaults set in Setup > General as they could change and if IDs are hidden by default, it could make the logged data less usable.

This PR changes the behavior so that the name order is always `realname firstname` and the ID is always shown when logging.

### CODE REVIEW

File: src/DbUtils.php [hunk 1]
```diff
@@ -1715,6 +1715,42 @@ public function getUserName($ID, $link = 0, $disable_anon = false)
...
+            'glpi_users',
+            [
+                'WHERE' => ['id' => $ID]
+            ]
+        );
+
+        if (count($iterator) === 1) {
+            $data     = $iterator->current();
+
+            if ($data['realname'] !== null && strlen($data['realname']) > 0) {
```
Comment (cedric-anne) [inline src/DbUtils.php:1739]:
```suggestion
            if (!empty($data['realname'])) {
```

File: src/DbUtils.php [hunk 2]
```diff
@@ -1715,6 +1715,42 @@ public function getUserName($ID, $link = 0, $disable_anon = false)
...
+                $formatted = $data['realname'];
+
+                if (strlen($data["firstname"]) > 0) {
```
Comment (cedric-anne) [inline src/DbUtils.php:1742]:
```suggestion
                if (!empty($data['firstname'])) {
```

**Review by orthagh — COMMENTED** [review]
Is a DBU function was required ?
a public function in `Log`  or better in `User` should be ok, no ?

**Comment by cconard96** [comment]
> Is a DBU function was required ? a public function in `Log` or better in `User` should be ok, no ?

It was only put in DBUtils since that is where the other `getUserName` function was. I've moved it to User now.

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by cedric-anne — COMMENTED** [review]
Tests have to be fixed.

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**